quickstart/kubernetes-guestbook.md

```diff
@@ -0,0 +1,373 @@
+                            // Kubernetes 1.3, DNS is a built-in service launched automatically. However, if the
+                            // cluster you are using does not have a built-in DNS service, you can instead access an
+                            // environment variable to find the master service's host. To do so, comment out the
+                            // 'value: "dns"' line above, and uncomment the line below:
+                            // value: "env",
+                        }],
+                        ports: [{ containerPort: 6379 }],
+                    }],
+                },
+            },
+        },
+    });
+
+    // FRONTEND
+    let frontendLabels = { app: "guestbook", tier: "frontend" };
+    let frontendService = new k8s.core.v1.Service("frontend", {
+        metadata: {
+            name: "frontend",
+            labels: frontendLabels,
+        },
+        spec: {
+            // Comment or delete the following line if you want to use a LoadBalancer:
+            type: "NodePort",
+            // If your cluster supports it, uncomment the following to automatically create
+            // an external load-balanced IP for the frontend service.
+            // type: "LoadBalancer",
+            ports: [{ port: 80 }],
+            selector: frontendLabels,
+        },
+    });
+    let frontendDeployment = new k8s.apps.v1.Deployment("frontend", {
+        metadata: { name: "frontend" },
+        spec: {
+            selector: { matchLabels: frontendLabels },
+            replicas: 3,
+            template: {
+                metadata: { labels: frontendLabels },
+                spec: {
+                    containers: [{
+                        name: "php-redis",
+                        image: "gcr.io/google-samples/gb-frontend:v4",
+                        resources: {
+                            requests: {
+                                cpu: "100m",
+                                memory: "100Mi",
+                            },
+                        },
+                        env: [{
+                            name: "GET_HOSTS_FROM",
+                            value: "dns",
+                            // Using `GET_HOSTS_FROM=dns` requires your cluster to provide a dns service. As of
+                            // Kubernetes 1.3, DNS is a built-in service launched automatically. However, if the
+                            // cluster you are using does not have a built-in DNS service, you can instead access an
+                            // environment variable to find the master service's host. To do so, comment out the
+                            // 'value: "dns"' line above, and uncomment the line below:
+                            // value: "env",
+                        }],
+                        ports: [{ containerPort: 80 }],
+                    }],
+                },
+            },
+        },
+    });
```

Comment:
We should possibly consider using the version of this code in https://github.com/pulumi/examples/tree/master/kubernetes-ts-guestbook instead of the version above.  It highlights more Pulumi concepts instead of being a literal port of the `.yaml` files, and is generally "nicer".

cc @hausdorff

Reply:
I looked at doing that, but it was a little too "advanced" for a getting started tutorial, IMHO. Plus, it involves multiple files, which just makes the overall tutorial a little bit harder to follow.

As you noted below, I alluded to building on this example to demonstrate the benefits of "real code" and I thought this would be a nice progression (basically, "quickstart 1a").